src/systems/destroy_entities.rs

```diff
@@ -0,0 +1,38 @@
+use amethyst::ecs::{Entities, System};
+use crossbeam_channel::Receiver;
+use types::EntityMessage;
+
+pub(crate) struct DestroyEntitiesSystem {
```

Comment:
Rather than having both a `CreateEntitiesSystem` and `DestroyEntitiesSystem`, I think it would make more sense to have a single system that manages both. With the way you have things setup now, both systems take incoming `EntityMessage` messages, but only handle one of the variants, meaning it could be possible to e.g. send `EntityMessage::Create` to `DestroyEntitiesSystem`, which would effectively be an error.